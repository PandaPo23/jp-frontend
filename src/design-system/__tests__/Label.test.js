import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Label from '../Label';

describe('Label component', () => {
  it('should render Label component', () => {
    const wrapper = mount(<Label>Label</Label>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render accessiblyHide Label', () => {
    const localProps = {
      hidden: true,
    };
    const wrapper = mount(<Label {...localProps}>Hidden Label</Label>);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(
      wrapper
        .find(Label)
        .first()
        .prop('hidden')
    ).toEqual(true);
    expect(
      wrapper
        .find(Label)
        .first()
        .prop('as')
    ).toBe('label');
    expect(wrapper.length).toBe(1);
  });
});
