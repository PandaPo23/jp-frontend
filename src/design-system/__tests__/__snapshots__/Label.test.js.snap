// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should render Label component 1`] = `
.emotion-0 {
  font-weight: bold;
  display: inline-block;
}

<Label
  as="label"
  display="inline-block"
  fontWeight="bold"
  hidden={false}
  nowrap={false}
>
  <label
    className="emotion-0"
  >
    Label
  </label>
</Label>
`;

exports[`Label component should render accessiblyHide Label 1`] = `
.emotion-0 {
  font-weight: bold;
  display: inline-block;
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

<Label
  as="label"
  display="inline-block"
  fontWeight="bold"
  hidden={true}
  nowrap={false}
>
  <label
    className="emotion-0"
  >
    Hidden Label
  </label>
</Label>
`;
